cirq/experiments/fidelity_estimation_test.py

```diff
@@ -306,6 +307,95 @@ def test_sample_2q_xeb_circuits_no_progress(capsys):
     assert captured.err == ''
 
 
+def _gridqubits_to_graph_device(qubits: Iterable[cirq.GridQubit]):
+    # cirq contrib: routing.gridqubits_to_graph_device
+    def _manhattan_distance(qubit1: cirq.GridQubit, qubit2: cirq.GridQubit) -> int:
+        return abs(qubit1.row - qubit2.row) + abs(qubit1.col - qubit2.col)
+
+    return nx.Graph(
+        pair for pair in itertools.combinations(qubits, 2) if _manhattan_distance(*pair) == 1
+    )
+
+
+def test_sample_2q_parallel_xeb_circuits():
+    circuits = rqcg.generate_library_of_2q_circuits(
+        n_library_circuits=5, two_qubit_gate=cirq.ISWAP ** 0.5, max_cycle_depth=10
+    )
+    cycle_depths = [10]
+    graph = _gridqubits_to_graph_device(cirq.GridQubit.rect(3, 2))
+    combs = rqcg.get_random_combinations_for_device(
+        n_library_circuits=len(circuits),
+        n_combinations=5,
+        device_graph=graph,
+        random_state=10,
+    )
+
+    df = sample_2q_xeb_circuits(
+        sampler=cirq.Simulator(),
+        circuits=circuits,
+        cycle_depths=cycle_depths,
+        combinations_by_layer=combs,
+    )
+    n_pairs = sum(len(c.pairs) for c in combs)
+    assert len(df) == len(cycle_depths) * len(circuits) * n_pairs
+    for (circuit_i, cycle_depth), row in df.iterrows():
+        assert 0 <= circuit_i < len(circuits)
+        assert cycle_depth in cycle_depths
+        assert len(row['sampled_probs']) == 4
+        assert np.isclose(np.sum(row['sampled_probs']), 1)
+        assert 0 <= row['layer_i'] < 4
+        assert 0 <= row['pair_i'] < 2  # in 3x2 graph, there's a max of 2 pairs per layer
+    assert len(df['pair_name'].unique()) == 7  # seven pairs in 3x2 graph
+
+
+def test_sample_2q_parallel_xeb_circuits_error():
+    circuits = rqcg.generate_library_of_2q_circuits(
+        n_library_circuits=5, two_qubit_gate=cirq.ISWAP ** 0.5, max_cycle_depth=10
+    )
+    cycle_depths = [10]
+    graph = _gridqubits_to_graph_device(cirq.GridQubit.rect(3, 2))
+    combs = rqcg.get_random_combinations_for_device(
+        n_library_circuits=len(circuits) + 100,  # !!! should cause invlaid input
+        n_combinations=5,
+        device_graph=graph,
+        random_state=10,
+    )
+
+    with pytest.raises(ValueError, match='.*invalid indices.*'):
+        _ = sample_2q_xeb_circuits(
+            sampler=cirq.Simulator(),
+            circuits=circuits,
+            cycle_depths=cycle_depths,
+            combinations_by_layer=combs,
+        )
+
+
+def test_sample_2q_parallel_xeb_circuits_error_2():
```

Comment:
please be more specific what kind of errors is that (as well in the above)

Reply:
changed both